### ISSUE TITLE
Report : Content - Article Category

### ISSUE THREAD

**Issue by micker** [joomla/accessibility#44]
**Elements must have sufficient color contrast**
#toolbar-help
To solve this violation, you need to:
Fix the following:
Element has insufficient color contrast of 3.04 (foreground color: #ffffff, background color: #17a2b8, font size: 11.3pt, font weight: normal). Expected contrast ratio of 4.5:1
Related node:
#toolbar-help

To solve this violation, you need to:
Fix the following:
Element has insufficient color contrast of 3.04 (foreground color: #ffffff, background color: #17a2b8, font size: 8.4pt, font weight: bold). Expected contrast ratio of 4.5:1
Related node:
.badge-info

**Form elements must have discembile text** 20 elements
Issue description
Ensures every form element has a label
Impact: critical
Element location
`<input type="checkbox" id="cb0" name="cid[]" value="2" onclick="Joomla.isChecked(this.checked);">`
To solve this violation, you need to:
Fix at least one (1) of these issues:
aria-label attribute does not exist or is empty
aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty or not visible
Form element does not have an implicit (wrapped) <label>
Form element does not have an explicit <label>
Element has no title attribute or the title attribute is empty

**Links must have discernible text**
Element location
`a[data-order="a\.lft"]`
Element source
`<a href="#" onclick="return false;" class="js-stools-column-order hasPopover" data-order="a.lft" data-direction="DESC" data-name="" title="" data-content="Select to sort by this column" data-placement="top" data-original-title="Ordering">`

To solve this violation, you need to:
Fix at least one (1) of these issues:
Element does not have text that is visible to screen readers
aria-label attribute does not exist or is empty
aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty or not visible
Element's default semantics were not overridden with role="presentation"
Element's default semantics were not overridden with role="none"
And
Fix the following:
Element is in tab order and does not have accessible text

**Checkbox inputs with the same name attribute value must be part of a group**
Ensures related `<input type="checkbox">` elements have a group and that the group designation is consistent
Impact: critical
Element location
`#cb0`
Element source
`<input type="checkbox" id="cb0" name="cid[]" value="2" onclick="Joomla.isChecked(this.checked);">`

To solve this violation, you need to:
Fix at least one (1) of these issues:
All elements with the name "cid[]" do not reference the same element with aria-labelledby
Element does not have a containing fieldset or ARIA group
Related nodes:
#cb1
#cb2
#cb3
#cb4
#cb5
#cb6
#cb7
#cb8
#cb9
#cb10
#cb11
#cb12
#cb13
#cb14
#cb15
#cb16
#cb17
#cb18
#cb19

**Form elements should have a visible label**
Ensures that every form element is not solely labeled using the title or aria-describedby attributes
Impact: serious
Element location
`input[name="checkall-toggle"]`
Element source
`<input type="checkbox" name="checkall-toggle" value="" class="hasTooltip" title="Check All Items" onclick="Joomla.checkAll(this)">`

To solve this violation, you need to:
Fix the following:
Only title used to generate label for form element

**The skip-link target should exist and be focusable**
Ensure all skip links have a focusable target

Impact: moderate
 Learn more
Element location
`a[href$="#collapse1"]`
Element source
`<a class="collapse-arrow" href="#collapse1"><span class="fa-fw fa fa-file-text"></span><span class="sidebar-item-title">Content</span></a>`

To solve this violation, you need to:
Fix the following:
No skip link target

**Comment by brianteeman** [comment]
is there really value in posting the results of automated testing tools?

**Comment by micker** [comment]
its a rewrited export all on necessary returns are killed ... what is the problem ? (i will to do better reporting for sure)

**Comment by brianteeman** [comment]
so if this is supposed to be a report of the page Content - Article Category?
You should only include in the report those issues that are specific to this page.
So for example the main landmark and the menu are generic to the site and not specific to this page and so obviously we dont need to see them in every report

It also really helps those who are going to be taking action on the report to see recommendations on the action to be taken. We all can run automated testing. The skill we lack, and what we are looking to the expertise of the a11y team for, is the understanding on what these reports mean in real terms. As you will I am sure agree some of the recommendations from the tool you are using make no sense at all in the context of joomla.

**Comment by micker** [comment]
ok i will update this with more specific return
no problem

**Comment by brianteeman** [comment]
The best example from the report of "bad" information is the part on the landmarks

**Comment by brianteeman** [comment]
> #toolbar-help

fixed with #21608

> Form elements must have discembile text 20 elements

already fixed

> Links must have discernible text

Fixed with https://github.com/joomla/joomla-cms/pull/21977 NEEDS tests

> Checkbox inputs with the same name attribute value must be part of a group

Fairly certain this is  false positive from the testing software 

> Form elements should have a visible label

Not a WCAG requirement. This is something only AXE reecommends

> The skip-link target should exist and be focusable

This is waiting on the menu to be finished
